packages/hardhat/contracts/Staking.sol

```diff
@@ -115,18 +147,34 @@ contract Staking is Ownable, ReentrancyGuard {
         if (userShares[user] == 0) return 0;
 
         // Calculate current exchange rate with accrued interest
-        uint256 currentExchangeRate = exchangeRate;
-        if (totalShares > 0) {
-            uint256 timeElapsed = block.timestamp - lastUpdateTime;
-            if (timeElapsed > 0) {
-                uint256 totalValue = (totalShares * exchangeRate) / PRECISION;
-                uint256 interest = (totalValue * interestRate * timeElapsed) / (SECONDS_PER_YEAR * 10000);
-                if (interest > 0) {
-                    currentExchangeRate += (interest * PRECISION) / totalShares;
-                }
-            }
-        }
+        uint256 currentExchangeRate = _getCurrentExchangeRate();
 
         return (userShares[user] * currentExchangeRate) / PRECISION;
     }
+
+    function _getCurrentExchangeRate() internal view returns (uint256) {
+        if (totalShares == 0) return exchangeRate;
+
+        uint256 timeElapsed = block.timestamp - lastUpdateTime;
+        if (timeElapsed == 0) return exchangeRate;
+
+        uint256 totalValue = (totalShares * exchangeRate) / PRECISION;
+        uint256 interest = (totalValue * interestRate * timeElapsed) / (SECONDS_PER_YEAR * 10000);
+
+        if (interest == 0) return exchangeRate;
+
+        return exchangeRate + ((interest * PRECISION) / totalShares);
+    }
+
+    // Function to check if additional token minting is needed
+    function additionalTokensNeeded() external view returns (uint256) {
```

Comment:
This feels like it is from a different attempt at solving the Staking contract balance issues. It doesn't make sense with the logic currently in MyUSD (assuming _enableVirtualBalance is enabled).